### PR TITLE
Revert "Avoid memset by using in-class member init."

### PR DESCRIPTION
**Pull request by marsupial**
Add comments as to why resetting to zero is necessary.

This reverts commit 65b4cd2cc3150b4d582338cf8920d20d1416681c.

### CODE REVIEW

File: include/cling/Interpreter/LookupHelper.h [hunk 1]
```diff
@@ -57,7 +57,7 @@ namespace cling {
   private:
     std::unique_ptr<clang::Parser> m_Parser;
     Interpreter* m_Interpreter; // we do not own.
-    const clang::Type* m_StringTy[kNumCachedStrings] = {};
+    const clang::Type* m_StringTy[kNumCachedStrings];
```
Comment (Axel-Naumann) [inline include/cling/Interpreter/LookupHelper.h]: I really don't want to leave this uninitialized. Too fragile; this is not C.

File: lib/Interpreter/LookupHelper.cpp [hunk 2]
```diff
@@ -1910,6 +1911,10 @@ namespace cling {
     assert(Type && "Type cannot be null");
     const Transaction*& Cache = m_Interpreter->getStdStringTransaction();
     if (!Cache || !m_StringTy[kStdString]) {
+      // getStringType can be called multiple times with Cache being null, and
+      // the local cache should be discarded when that occurs.
+      if (!Cache) ::memset(m_StringTy, 0, sizeof(m_StringTy));
```
Comment (Axel-Naumann) [inline lib/Interpreter/LookupHelper.cpp]: Can we leave this line (and this comment) but rephrase it as
```
m_StringTy = {};
```
That should be fine, even if leaving the in-class initializer, or am I overlooking something?

**Comment by marsupial** [comment]
You can adjust however you like, but I tried m_StringTy = {}, and clang complained it was not a legal statement.  I don't have any particular care as to how the m_StringTy data gets back to zero, just that it does when Cache is null.

I'm a little against the in class init, just because it is a non trial amount of data that doesn't need to be initialized if it is never used, but understand that -appearing- safer is not a bad thing.

Up to you.

**Comment by Axel-Naumann** [comment]
Thanks! And indeed, that'll need to become a `std::array`. I will take care of it and then close this (unless you have already provided an update PR when I get up tomorrow ;-)

**Comment by marsupial** [comment]
Probably not me, I'll be busy dancing with conflicts from deb1fcb !

**Comment by Axel-Naumann** [comment]
Thanks for explaining! Solved differently, see aae2870f62177c7ad82143dabbb97b02c30f80b2

**Pull request closed without merging**